match/Repetitions.roc

```diff
@@ -0,0 +1,80 @@
+interface Repetitions
+    exposes [
+        match,
+    ]
+    imports []
+
+match : Str, Str -> Bool
+match = \pattern, text ->
+    # convert pattern and text from Str to List U8
+    patternList = Str.toUtf8 pattern
+    textList = Str.toUtf8 text
+
+    # if the pattern begins with a ^, we'll only check for matches at the beginning of `text`
+    if List.first patternList == Ok '^' then
```

Comment:
I think it would be slightly more idiomatic Roc to pattern match on the list instead of getting the first element and using an if else.